pkg/kudoctl/clog/log.go

```diff
@@ -0,0 +1,131 @@
+package clog
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/spf13/pflag"
+)
+
+// this package provides (C)lient logs, or CLI logs. Although this is called a log, it is for writing to STDOUT and is designed
+// for CLI output (not to log files etc.). We want -V behavior familiar to kubectl users, but klog comes with a lot of dependencies
+// and global flags that obfuscate KUDO.  glog is built more for a multi-threaded logger (managing buffers etc) which
+// KUDO CLI doesn't have a need for.  clog provides verbosity control for CLI output.
+
+// guidance for use of V level
+//  0-1 normal standard out
+//  2-4 as debug-level logs
+//  5-6 logical chooses
+//  7-8 input/output details
+//  9-10 as trace-level (http details)
+
+// Level specifies a level of verbosity for V logs.
+type Level int32
+
+// get returns the value of the Level.
+func (l *Level) get() Level {
+	return *l
+}
+
+// set sets the value of the Level.
+func (l *Level) set(val Level) {
+	atomic.StoreInt32((*int32)(l), int32(val))
+}
+
+// Get is part of the flag.Value interface.
+func (l *Level) Get() interface{} {
+	return *l
+}
+
+// String is part of the flag.Value interface.
+func (l *Level) String() string {
+	return strconv.FormatInt(int64(*l), 10)
+}
+
+// Set is part of the flag.Value interface.
+func (l *Level) Set(value string) error {
```

Comment:
I dont see where this actually gets called. Is this not fully being used yet?

Reply:
the important part here is L48 comment above this line... https://golang.org/pkg/flag/#example_Value
In our case we are using pflag... which requires the following: https://godoc.org/github.com/spf13/pflag#Value

```type Value interface {
    String() string
    Set(string) error
    Type() string
}```

Reply:
:+1: thank you!

Reply:
that's super clean, I like it. I should've read the flag.Value interface, being sick is catching up with me.